Projects/UOContent/Mobiles/BaseCreature.cs

```diff
@@ -4312,12 +4312,15 @@ public virtual bool CheckFeed(Mobile from, Item dropped)
                 }
                 else if (m_Loyalty < MaxLoyalty)
                 {
-                    // 50% chance to increase 10 loyalty per food
-                    m_Loyalty = Math.Min(MaxLoyalty, Utility.CoinFlips(amount, MaxLoyaltyIncrease) * 10);
-                }
+                    // Calculate the loyalty increase
+                    int loyaltyIncrease = Utility.CoinFlips(amount, MaxLoyaltyIncrease) * 10;
 
-                // looks like in OSI pets say they are happier even if they are at maximum loyalty
-                SayTo(from, 502060); // Your pet looks happier.
+                    if (loyaltyIncrease > 0)  // Only update if there's an actual increase
```

Comment:
On OSI, even if loyalty doesn't go up, it still says your pet looks happier. I am ok deviating though.